File: scala/src/main/scala/com/shekhargulati/ninetynine_problems/scala/arithmetic/P33.scala
package com.shekhargulati.ninetynine_problems.scala.arithmetic

import com.shekhargulati.ninetynine_problems.scala.lists.P10

/**
  * (**) Determine the prime factors of a given positive integer (2)
  */
object P33 {

  def primeFactorsMult(number: Int): List[(Int, Int)] = P10.encode(P32.primeFactors(number)).map { case (t1, t2) => (t2, t1) }

}

File: haskell/02-arithmetic/P33.hs
module P33 where
import P32
import P10
import Data.Tuple
-- (**) Determine the prime factors of a given positive integer (2).

primeFactorsMult :: Int -> [(Int, Int)]
primeFactorsMult number = map (\t -> swap t) (encode (primeFactors number))
